src/main/kotlin/no/nav/familie/ef/sak/behandling/oppgaveforopprettelse/OppgaverForOpprettelseService.kt

```diff
@@ -44,13 +48,31 @@ class OppgaverForOpprettelseService(
         if (saksbehandling.stønadstype != StønadType.OVERGANGSSTØNAD) {
             return emptyList()
         }
-        val tilkjentYtelse = tilkjentYtelseService.hentForBehandlingEllerNull(behandlingId)
-        val kanOppretteInntektskontroll = kanOppretteOppgaveForInntektskontrollFremITid(tilkjentYtelse)
 
+        val vedtak = vedtakService.hentVedtak(behandlingId)
+        val kanOppretteInntektskontroll = if (vedtak.resultatType == ResultatType.AVSLÅ &&
+            vedtak.avslåÅrsak == AvslagÅrsak.MINDRE_INNTEKTSENDRINGER
+        ) {
+            kanOppretteInntektsoppgaveForSisteIverksatteBehandling(saksbehandling.fagsakId)
+        } else if (vedtak.resultatType == ResultatType.INNVILGE) {
+            val tilkjentYtelse = tilkjentYtelseService.hentForBehandlingEllerNull(behandlingId)
+            kanOppretteOppgaveForInntektskontrollFremITid(tilkjentYtelse)
+        } else {
+            false
+        }
         return if (kanOppretteInntektskontroll) listOf(OppgaveForOpprettelseType.INNTEKTSKONTROLL_1_ÅR_FREM_I_TID) else emptyList()
     }
 
-    fun initialVerdierForOppgaverSomSkalOpprettes(behandlingId: UUID) = hentOppgavetyperSomKanOpprettes(behandlingId)
+    private fun kanOppretteInntektsoppgaveForSisteIverksatteBehandling(behandlingId: UUID): Boolean {
```

Comment:
Kan denne endres til hentSisteTilkjentYtelse - så kan du gjøre selve `kanOppretteOpppgaverForInntektskontrollFremITid(tilkjentYtelse)` i den forrige metoden?

Da er det "like" løyper i den if/elsen - begge henter ut en tilkjentytelse og sjekker om det kan opprettes inntektsoppgaer for det..?

Reply:
Ja, endret til "like løyper" i den metoden 👍